### PR TITLE
SEND module fixes & Stuff

### PR DESCRIPTION
**Pull request by gpotter2**
Quite weird PR, every thing is related to the `send`'s module fix
- Fixes SEND module (+ Make it work + tests)
- Fix RandBin on python3 (wrongly created)
- Remove the `get_cls` usages from inet6 and use direct classes instead: this allow us to inject the class from `contrib/send.py` into inet6. It was not possible because of `globals()` restrictions.
- Minor new log in rdpcap

Fixes https://github.com/secdev/scapy/issues/1045

### CODE REVIEW

File: scapy/volatile.py [hunk 1]
```diff
@@ -252,7 +252,12 @@ def __mul__(self, n):
 
 class RandBin(RandString):
     def __init__(self, size=None):
-        RandString.__init__(self, size, "".join(map(chr, range(256))))
+        RandString.__init__(self, size, b"".join(map(chb, range(256))))
```
Comment (p-l-) [inline scapy/volatile.py]: I think it would be better to use `super()` here. Also, we could remove `map()`, and we should name the named arguments:
```python
    def __init__(self, size=None):
        super(RandBin, self).__init__(size=size, chars=b"".join(chb(c) for c in range(256)))
```

File: scapy/volatile.py [hunk 2]
```diff
@@ -250,9 +250,14 @@ def _fix(self):
     def __mul__(self, n):
         return self._fix()*n
 
-class RandBin(RandString):
+class RandBin(RandString, object):
```
Comment (p-l-) [inline scapy/volatile.py]: I think a "string" in Scapy's spirit (if such a thing exists) is actually a `bytes` object in Python 3 (see `Str*Field`s). So maybe `RandString._fix()` should return a `bytes` object. What do you think?

Comment (gpotter2) [reply scapy/volatile.py]: Not sure about this, but i'm more to keep current behavior because:
- many other volatile values that are returning str (definatly not bytes), such as RandIP, RandMAC, are subclasses of RandString. It would make no sense if those returned `str` and RandStr `bytes`
- RandBin already exists

Comment (p-l-) [reply scapy/volatile.py]: A string, in Python 3, means a unicode string. A RandString must return bytes.

Comment (gpotter2) [reply scapy/volatile.py]: Got it. Should `RandIP` still inherit from `RandString` ?

File: test/regression.uts [hunk 3]
```diff
@@ -8716,7 +8716,7 @@ assert(rss == ("CON:" if six.PY2 else "foo.exe:"))
 
 random.seed(0x2807)
 rts = RandTermString(4, "scapy")
-assert(sane(raw(rts)) in ["...[scapy", "......scapy"])
+assert(sane(raw(rts)) in ["...Zscapy", "$#..scapy"])
```
Comment (gpotter2) [inline test/regression.uts]: One will notice that current implementation is broken:
e.g. on python 3, the random string should have a length of 4, + "scapy".
`"......scapy"` is wrong

File: scapy/arch/__init__.py [hunk 4]
```diff
@@ -25,7 +25,7 @@ def str2mac(s):
         from scapy.arch.bpf.core import get_if_raw_addr
 
 def get_if_addr(iff):
-    return socket.inet_ntoa(get_if_raw_addr(iff))
+    return inet_ntop(socket.AF_INET, get_if_raw_addr(iff))
```
Comment (gpotter2) [inline scapy/arch/__init__.py]: This is not really related. However, I had a bug with this while testing.
`inet_ntoa` is getting deprecated, so I'm switching them to `inet_ntop` when scapy bug on them

**Comment by guedou** [comment]
Tests are failing on Python3. I think that the RandBin could be used for #1168.

File: scapy/utils.py [hunk 5]
```diff
@@ -852,7 +852,12 @@ def read_all(self,count=-1):
         res=[]
         while count != 0:
             count -= 1
-            p = self.read_packet()
+            try:
+                p = self.read_packet()
+            except:
+                if conf.debug_dissector:
```
Comment (p-l-) [inline scapy/utils.py]: That's not how `conf.debug_dissector` is used for. It is used to let exceptions happen when dissecting a packet, rather than using `Raw()` instead.

Why do you need this change?

Comment (gpotter2) [reply scapy/utils.py]: Was used while debugging. Removed

**Comment by guedou** [comment]
@gpotter2 @p-l- is this PR ready to be merged?
